src/discretize.jl

```diff
@@ -217,20 +217,20 @@ function generate_training_sets(domains, dx, eqs, bcs, eltypeθ, dict_indvars::D
 
     bcs_train_sets = map(bound_args) do bt
         span = map(b -> get(dict_var_span, b, b), bt)
-        _set = adapt(eltypeθ,
+        _set = adapt(Array{eltypeθ},
```

Comment:
Do not force an `Array` type, since that won't work in most scenarios. Just add a post-adapt `convert.(eltypeθ, x)`